src/processing/processing_subtask_queue_manager.cpp

```diff
@@ -460,23 +669,144 @@ namespace sgns::processing
         }
     }
 
+    bool ProcessingSubTaskQueueManager::HasAvailableWork(bool checkOwnershipQuota) const
+    {
+        // Check if we've already processed the maximum allowed subtasks per ownership
+        if ( checkOwnershipQuota && m_processedSubtasksInCurrentOwnership >= m_maxSubtasksPerOwnership)
```

Comment:
I'd add a parenthesis 
```suggestion
        if ( checkOwnershipQuota && (m_processedSubtasksInCurrentOwnership >= m_maxSubtasksPerOwnership) )
```